src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1093,115 +1208,228 @@ private static Symbol FindMostSpecificImplementation(Symbol interfaceMember, Nam
             out Symbol conflictingImplementation1,
             out Symbol conflictingImplementation2)
         {
-            var implementations = ArrayBuilder<(MultiDictionary<Symbol, Symbol>.ValueSet MethodSet, MultiDictionary<NamedTypeSymbol, NamedTypeSymbol> Bases)>.GetInstance();
+            ImmutableArray<NamedTypeSymbol> allInterfaces = implementingType.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics);
+
+            if (allInterfaces.IsEmpty)
+            {
+                conflictingImplementation1 = null;
+                conflictingImplementation2 = null;
+                return null;
+            }
+
+            // Properties or events can be implemented in an unconventional manner, i.e. implementing accessors might not be tied to a property/event.
+            // If we simply look for a more specific implementing property/event, we might find one with not most specific implementing accessors.
+            // Returning a property/event like that would be incorrect because runtime will use most specific accessor, or it will fail because there will
+            // be an ambiguity for the accessor implementation.
+            // So, for events and properties we look for most specific implementaion of corresponding accessors and then try to tie them back to
+            // an event/property, if any.
+            (MethodSymbol interfaceAccessor1, MethodSymbol interfaceAccessor2) = GetImplementableAccessors(interfaceMember);
+
+            if (interfaceAccessor1 is null && interfaceAccessor2 is null)
+            {
+                return findMostSpecificImplementationInBases(interfaceMember, allInterfaces, ref useSiteDiagnostics, out conflictingImplementation1, out conflictingImplementation2);
+            }
+
+            Symbol accessorImpl1 = findMostSpecificImplementationInBases(interfaceAccessor1 ?? interfaceAccessor2, allInterfaces, ref useSiteDiagnostics,
```

Comment:
>interfaceAccessor1 ?? interfaceAccessor2 [](start = 73, length = 40)

I didn't understand this. In the case where both accessors are non-null, wouldn't we have to look both of them up in bases, to find the most specific implementation?